### PR TITLE
Table custom columns content.

### PR DESCRIPTION
**Pull request by larskendall**

### CODE REVIEW

File: src/components/pages/roster-page/roster-page.tsx [hunk 1]
```diff
@@ -147,9 +72,33 @@ export const RosterPage = () => {
...
+  //
 
-  function handleFileInputChange(e: ChangeEvent<HTMLInputElement>) {
+  const handleChangePage = (event: MouseEvent<HTMLButtonElement> | null, pageNew: number) => {
+    setPage(pageNew);
+  };
+
+  const handleChangeRowsPerPage = (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setRowsPerPage(parseInt(event.target.value));
+    setPage(0);
```
Comment (jdgarrett) [inline src/components/pages/roster-page/roster-page.tsx:98]: Can we calculate the page based on which page the first row of the table is going to be on?  For example, if they were on page 6 showing 10 rows per page, they should be on page 2 if they switch to 50 rows per page, or page 3 if they switch to 25 rows per page.

File: src/components/pages/roster-page/roster-page.tsx [hunk 2]
```diff
@@ -169,20 +118,9 @@ export const RosterPage = () => {
           message: `Successfully uploaded ${count} roster entries.`,
           onClose: () => { setAlertDialogProps({ open: false }); },
         });
-        initializeTable();
+        initializeTable().then();
```
Comment (jdgarrett) [inline src/components/pages/roster-page/roster-page.tsx:121]: Can this be awaited?

File: src/components/tables/table-custom-columns-content.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            {columns.map(column => (
+              <TableCell key={`${column.name}-${row.edipi}`}>
+                {row[column.name]}
+              </TableCell>
+            ))}
+
+            {showActions() && (
+              <TableCell className={classes.tableButtons}>
+                <Button
+                  className={classes.editRosterEntryButton}
```
Comment (jdgarrett) [inline src/components/tables/table-custom-columns-content.tsx:62]: Minor nitpick, but I would remove roster from these names.

File: src/components/tables/table-custom-columns-content.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+          {showActions() && (
+            <TableCell />
+          )}
+        </TableRow>
+      </TableHead>
+
+      <TableBody>
+        {rows.map(row => (
+          <TableRow key={`${row.edipi}`}>
```
Comment (jdgarrett) [inline src/components/tables/table-custom-columns-content.tsx:52]: We can't assume that `edipi` will exist in the row.  In fact, EDIPI will not be present for non-PII views.

File: src/components/tables/table-custom-columns-content.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            {columns.map(column => (
+              <TableCell key={`${column.name}-${row.edipi}`}>
```
Comment (jdgarrett) [inline src/components/tables/table-custom-columns-content.tsx:54]: `edipi` reference.

**Review by jdgarrett — CHANGES_REQUESTED** [review]

File: src/components/pages/roster-page/roster-page.tsx (hunk 1 above)
Comment (larskendall) [reply src/components/pages/roster-page/roster-page.tsx:98]: Yeah I think that would be better UX for sure.

File: src/components/tables/table-custom-columns-content.tsx (hunk 3 above)
Comment (larskendall) [reply src/components/tables/table-custom-columns-content.tsx:62]: Oops

File: server/api/roster/roster.controller.ts [hunk 6]
```diff
@@ -164,13 +164,12 @@ class RosterController {
       })
       .getRawMany<RosterEntryData>();
 
-    res.json(roster);
-  }
-
-  async getRosterCount(req: ApiRequest<OrgParam>, res: Response) {
-    const count = await (await queryAllowedRoster(req.appOrg!, req.appRole!)).getCount();
+    const totalRowsCount = await (await queryAllowedRoster(req.appOrg!, req.appRole!)).getCount();
```
[the remark this answers is not in the document]
Comment (larskendall) [reply server/api/roster/roster.controller.ts:167]: Hmm this seems like an edge case that might not be worth complicating the code with. That would only be the situation on the very last page of the roster right? I'd rather avoid any early optimizations that aren't going to make an obvious impact in performance.

**Pull request merged**